From 231eff875b07ac2e20125fac07f87e5d7863baf7 Mon Sep 17 00:00:00 2001
From: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Date: Fri, 26 Oct 2012 16:29:20 +0200
Subject: [PATCH 43/78] wifi: prevent add interface while calibrating

BZ: 61194

At first boot, it is possible to have the Setup Wizard turning
wifi ON in parallel with the Wifi chip calibration process.
This can lead to unexpected behavior

Change-Id: Iebafae0fdb04fddd29fc43423bc27f9034003127
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Reviewed-on: http://android.intel.com:8080/71894
Reviewed-by: buildbot <buildbot@intel.com>
Reviewed-by: Tricot, HonoreX <honorex.tricot@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Tested-by: lab_cactus <lab_cactus@intel.com>
Reviewed-on: http://android.intel.com:8080/78000
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wl12xx-compat/drivers/net/wireless/wl12xx/main.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
index fdd9139..9196d9d 100644
--- a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
+++ b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
@@ -2374,6 +2374,11 @@ static int wl1271_op_start(struct ieee80211_hw *hw)
 
 	wl1271_debug(DEBUG_MAC80211, "mac80211 start");
 
+	if (wl->state == WL1271_STATE_PLT) {
+		wl1271_warning("start not possible while calibrating");
+		return -EBUSY;
+	}
+
 	/*
 	 * We have to delay the booting of the hardware because
 	 * we need to know the local MAC address before downloading and
-- 
1.7.9

